plugin/trino-sqlserver/src/test/java/io/trino/plugin/sqlserver/TestingSqlServer.java

```diff
@@ -93,7 +93,10 @@ public TestingSqlServer(String version, BiConsumer<SqlExecutor, String> database
         databaseName = initializedState.databaseName;
         cleanup = initializedState.cleanup;
 
-        container.withUrlParam("database", databaseName);
+        container.withUrlParam("database", databaseName)
+                // Instead of having a statement block forever (the default)
+                // Throws SQLServerException with SQLServerError code 1222  "Lock request time out period exceeded"
+                 .withUrlParam("lockTimeout", Integer.toString(120*1000));
```

Comment:
Not strictly necessary, but it was nice for me to have tests fail instead of hang forever. 

For example, when I purposefully tried constructing a deadlocking situation but messed the order of operations up the transaction was blocked forever instead of becoming deadlocked and killed.

Reply:
Can you create a PR with just this change and stress-test (add multiple runs of trino-sqlserver tests in ci.yml). It would make it easier to find all occurrences of such deadlocks that happen during tests.

Reply:
Sure will break this out into a separate pr.

Reply:
PR here: https://github.com/trinodb/trino/pull/10100